### ISSUE TITLE
Error pulling docker image

### ISSUE THREAD

**Issue by cromelex** [rishikanthc/markopolis#4]
I wanted to spin this up to give it a try but I am getting an error pulling the docker image.

Would you be able to check if the correct version is published?

Thanks

**Comment by rishikanthc** [comment]
can you give me the error message ? any details you can share would be of help. This should be trivial to fix

**Comment by rishikanthc** [comment]
Turns out my package visibility settings was set to private. This should be fixed now. Can you try again and let me know if this is resolved ?

**Comment by cromelex** [comment]
Cheers. That fixed it.

So initially I've deployed this with the environment variables on a `.env` (out of habit with most of my selfhosted stuff).

MARKOPOLIS_DOMAIN="https://your-domain.com" works, but I don't think the rest of the variables were picked up.

Instead, it deployed a copy of your demo site. I am not sure if this is supposed to have been included. You might want to remove it from the docker container
The API key on the .env was not picked up, so when i went to localhost:8080/docs and used the API key it throws an invalid error key.


But other than that I managed to get it up and running, seeing a couple of issues but I will raise separately to make your life easier.

**Comment by cromelex** [comment]
Can I double check 2 things?

1st, the "index" file must always be  `home.md`, lowercase, correct? I cannot make it, say, `Home.md` just for the sake of it being capitalised in the links etc on the lefthandside?

2nd, the proper way to use this would be to, say,

- mount a volume with the source files (ie, the Obsidian vault, as an example).
- from within the docker container, navigate to said folder and use `consume`

Is this correct?

Also, removal is not yet implemented?

Thank you in advance.

**Comment by rishikanthc** [comment]
Regarding 1st,
Yes that's correct. If you want to make the links to be capitalized, set the title filed in the front matter. That's picked up. The filename is used only if the title is not set in the front matter.

Regarding 2,
You don't need to mount a volume with the source files. The consume command is meant to be run locally on your computer. The docker volume is just for persistent storage to store the markdown files for the server.
On your local system where you have the md files, install the python package. setup a config.yaml and then run consume inside the root folder where md files are present. the consume command will connect to the server and upload all md files.

And yes removal is currently not implemented, so you won't be able to remove. I plan to implement this today. Lookout for a release sometime later in the day.

In summary, the consume command is a convenience command to run locally. You don't need to manually upload your source files or deal with docker at all. 
Hope this helps.

**Comment by rishikanthc** [comment]
Just noticed your previous comment. Reopening the issue. I'll debug this now. Another user seems to be facing the same issue with env variables not being picked up.

**Comment by cromelex** [comment]
> Regarding 1st, Yes that's correct. If you want to make the links to be capitalized, set the title filed in the front matter. That's picked up. The filename is used only if the title is not set in the front matter.
> 

I've set this as `title: Home` and the title of the page changes - the link on the left does not. If you could make it so it's changed on the left as well that would be great
![image](https://github.com/user-attachments/assets/d43007da-0576-43a1-b725-e99fbd55adf3)



> Regarding 2, You don't need to mount a volume with the source files. The consume command is meant to be run locally on your computer. The docker volume is just for persistent storage to store the markdown files for the server. On your local system where you have the md files, install the python package. setup a config.yaml and then run consume inside the root folder where md files are present. the consume command will connect to the server and upload all md files.
> 
> In summary, the consume command is a convenience command to run locally. You don't need to manually upload your source files or deal with docker at all. Hope this helps.

Cheers. Thanks for clarifying. I am not familiar with python packages, so I hadn't gotten that far.
On my current setup my server/client are the same machine so it was easier just do the docker volume mount. Understand the server/local config makes a lot of sense for the scenario where they are not the same machine.

**Comment by rishikanthc** [comment]
This has been resolved in the latest release :)

**Comment by cromelex** [comment]
> Oops. I'll fix that so the link on the left also changes. Keep on eye out for an update. I'll leave this issue open for reference.

> This has been resolved in the latest release :)

Looking good, but can I suggest you make it so the folder on the left (I'll call it the menu) stays expanded when you are navigating within any of the pages contained in that folder?

**Comment by rishikanthc** [comment]
Sure. I was gonna do that because I myself found it annoying to keep going back. I'm in the middle of rewriting the core API. Once I finish that I'll work on adding this.

**Comment by cromelex** [comment]
It is working fine but I'm just letting you know since this is the first time I've seen this docker warning. Only happened with the latest version when pulling.

>  ! markopolis The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64/v4) and no specific platform was requested    0.0s
